src/k8s/pkg/client/kubernetes/status.go

```diff
@@ -20,6 +21,25 @@ func (c *Client) WaitKubernetesEndpointAvailable(ctx context.Context) error {
 	})
 }
 
+// IsClusterReady pings /readyz endpoint and only reports ready if status code == 200.
+func (c *Client) IsClusterReady(ctx context.Context) bool {
+	logger := log.FromContext(ctx)
+	result := c.Discovery().RESTClient().Get().AbsPath("/readyz").Do(ctx)
+	if result.Error() != nil {
+		logger.Info("Failed to check /readyz endpoint", "error", result.Error())
```

Comment:
this could easily flood the logs with log lines that are completely out of context, same below.

If you think this is a useful log line, maybe make this a `.V(1)` or `.V(2)`